pkg/operator/controllers/checker/subnets.go

```diff
@@ -0,0 +1,63 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	maoclient "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	azureproviderv1beta1 "sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+)
+
+func getSubnetIDs(ctx context.Context, vnetID string, clustercli maoclient.Interface) (masterSubnetID string, workerSubnetIDs []string, err error) {
+
+	machines, err := clustercli.MachineV1beta1().Machines(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return masterSubnetID, workerSubnetIDs, err
```

Comment:
```suggestion
		return "", "", err
```

always write ^ to make it easier for the reader to reason that "if an error is returned, the other values are zero values"